Framework/Algorithms/src/ConvertToHistogram.cpp

```diff
@@ -42,7 +42,7 @@ bool ConvertToHistogram::isWorkspaceLogical(
     const MatrixWorkspace_sptr inputWS) const {
```

Comment:
I think you can actually remove this method from all involved classes. The check in this method is not necessary anymore, since `Histogram` guarantees that this is always true.